### PR TITLE
Create custom onsite payments app extension template

### PR DESCRIPTION
**Pull request by marklevi**
### Background

Related issue: https://github.com/Shopify/payments-platform/issues/4381
Related project: https://docs.google.com/document/d/1Y2Ly5sw6MalZQP8w3yraQlOs8dK3UK38LVQvQz7fC_M/edit#heading=h.8coln95uazox
Related management class: [CustomOnsite management class](https://github.com/Shopify/shopify/blob/main/components/payment_processing/payments_partners/app/models/payments_partners/payments_app_extensions/custom_onsite_payments_app_extension.rb)

Creates an custom onsite payments app extension template to be used via the CLI

(Provide a link to any relevant issues AND provide a TLDR of the issue in this pull request)
### Checklist

- [ ] I have :tophat:'d these changes
- [x] I have squashed my commits into chunks of work with meaningful commit messages

### CODE REVIEW

File: payments-app-extension-custom-onsite/shopify.extension.toml.liquid [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+start_void_session_url = "https://api.paymentprovider.com/void"
+supported_countries = ["US"]
+supported_payment_methods = ["visa"]
+supports_3ds = false
+supports_installments = false
+supports_deferred_payments = false
+test_mode_available = true
+api_version = "2021-07"
+multiple_capture = false
+update_payment_session_url = "https://api.paymentprovider.com/update"
```
Comment (wikiti) [inline payments-app-extension-custom-onsite/shopify.extension.toml.liquid:24]: I don't see this field in the partners UI. What's this?

File: payments-app-extension-custom-onsite/shopify.extension.toml.liquid [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ui_extension_registration_uuid = "0e63f1a6-ed7f-41b2-8e56-e6399ac48764"
+checkout_payment_method_fields = []
+modal_payment_method_fields = []
```
Comment (wikiti) [inline payments-app-extension-custom-onsite/shopify.extension.toml.liquid:27]: I can only see one of this fields in the partners UI (UI Extension Field Definitions) 🤔 

image

File: payments-app-extension-custom-onsite/shopify.extension.toml.liquid (hunk 1 above)
Comment (marklevi) [reply payments-app-extension-custom-onsite/shopify.extension.toml.liquid:24]: The field exists in the management class: https://github.com/Shopify/shopify/blob/main/components/payment_processing/payments_partners/app/models/payments_partners/payments_app_extensions/custom_onsite_payments_app_extension.rb#L49

File: payments-app-extension-custom-onsite/shopify.extension.toml.liquid (hunk 2 above)
Comment (marklevi) [reply payments-app-extension-custom-onsite/shopify.extension.toml.liquid:27]: Both fields are included here: 
https://github.com/Shopify/shopify/blob/main/components/payment_processing/payments_partners/app/models/payments_partners/payments_app_extensions/custom_onsite_payments_app_extension.rb#L51

**Pull request merged**